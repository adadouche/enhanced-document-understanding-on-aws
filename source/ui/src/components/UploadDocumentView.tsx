/**********************************************************************************************************************
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.                                                *
 *                                                                                                                    *
 *  Licensed under the Apache License, Version 2.0 (the License). You may not use this file except in compliance      *
 *  with the License. A copy of the License is located at                                                             *
 *                                                                                                                    *
 *      http://www.apache.org/licenses/LICENSE-2.0                                                                    *
 *                                                                                                                    *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES *
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions    *
 *  and limitations under the License.                                                                                *
 *********************************************************************************************************************/
import {
    Alert,
    AppLayout,
    Button,
    Container,
    ContentLayout,
    Form,
    FormField,
    Header,
    ProgressBar,
    Select,
    SelectProps,
    SpaceBetween
} from '@cloudscape-design/components';
import { generateToken, getUsername } from './DocumentTable/DocumentTable';
import { FileSize, FileType } from './FileUpload/interfaces';

import { API } from 'aws-amplify';
import React from 'react';
import { useNavigate } from 'react-router-dom';
import { API_NAME, MAX_UPLOAD_FILE_SIZE } from '../utils/constants';
import FileUpload from './FileUpload/FileUpload';
import { mapResultsToCases } from './makeData';
import { formatFileSize } from './FileUpload/internal';

export interface WorkflowConfig {
    NumRequiredDocuments: number;
    UniqueDocumentTypes: string[];
    WorkflowConfigName: string;
}

type UploadDocumentViewProps = {
    caseName?: string;
    caseId?: string;
    casesList: any[];
    setCasesList: Function;
    workflowConfig: WorkflowConfig;
    selectedCaseRemainingRequiredDocs: any;
};

type Case = {
    name: string;
    status: string;
    caseDocuments: Object[];
    dateCreated: string;
};

type CaseMap = {
    [key: string]: Case;
};

let token: string;
let results: any;

enum Status {
    IN_PROGRESS = 'in-progress',
    SUCCESS = 'success',
    ERROR = 'error'
}

export default function UploadDocumentView(props: UploadDocumentViewProps) {
    let caseMap: CaseMap = {};
    let casesOptions: SelectProps.Option[] = [];

    const updateCaseMap = () => {
        if (props.casesList) {
            for (const item of props.casesList) {
                const _case: Case = {
                    name: item.name,
                    status: item.status,
                    caseDocuments: item.caseDocuments,
                    dateCreated: item.dateCreated
                };
                caseMap[item.caseId] = _case;
            }
        }
        for (const caseId in caseMap) {
            casesOptions.push({
                label: caseMap[caseId].name,
                value: caseId,
                description: `Case Id: ${caseId}`,
                disabled:
                    caseMap[caseId].status !== 'initiate' ||
                    caseMap[caseId].caseDocuments.length >= props.workflowConfig.NumRequiredDocuments
            });
        }
    };

    updateCaseMap();

    let caseId: string | undefined;
    if (props.caseId) {
        caseId = props.caseId;
    }

    const [initialSelectedCaseId] = React.useState<string>(caseId!);
    const [initialSelectedCaseName] = React.useState<string>(caseMap[initialSelectedCaseId]?.name);
    const [selectedCaseOption, setSelectedCaseOption] = React.useState<SelectProps.Option>({
        label: initialSelectedCaseName,
        value: initialSelectedCaseId
    });
    const [chosenFiles, setChosenFiles] = React.useState<File[]>([]);
    const [showProgressBar, setShowProgressBar] = React.useState(false);
    const [progressStatus, setProgressStatus] = React.useState<Status | undefined>();
    const [filesFailedToUpload, setFilesFailedToUpload] = React.useState<string[]>([]);
    const [uploadResultMessage, setUploadResultMessage] = React.useState('');
    const [uploadProgress, setUploadProgress] = React.useState(0);
    const [chooseFilesError, setChooseFilesError] = React.useState('');
    const [allFilesAttempted, setAllFilesAttempted] = React.useState(false);
    const [documentTypeList, setDocumentTypeList] = React.useState<string[]>([]);
    const [refreshCaseData, setRefreshCaseData] = React.useState(false);
    const [selectedCaseRemainingAllowedDocs, setSelectedCaseRemainingAllowedDocs] = React.useState();
    const navigate = useNavigate();

    const fileMetadata = { name: true, type: true, size: FileSize.MB, lastModified: true, thumbnail: true };
    const allowMultipleFiles = true;

    React.useEffect(() => {
        const newRemainingDocs = Object.assign({}, props.selectedCaseRemainingRequiredDocs);

        for (let docType of documentTypeList) {
            if (newRemainingDocs[docType]) {
                newRemainingDocs[docType] -= 1;
            }
        }
        setSelectedCaseRemainingAllowedDocs(newRemainingDocs);
    }, [documentTypeList, props.selectedCaseRemainingRequiredDocs]);

    React.useEffect(() => {
        const fetchData = async () => {
            token = await generateToken();
            results = await API.get(API_NAME, 'cases', {
                headers: {
                    Authorization: token
                }
            });
            const newData = mapResultsToCases(results.Items);
            props.setCasesList(newData);
        };
        if (!props.casesList || refreshCaseData) {
            fetchData();
            updateCaseMap();
            setRefreshCaseData(false);
        }
    }, [refreshCaseData]);

    const handleFileSelection = async (newFiles: FileType) => {
        setChooseFilesError('');
        if (newFiles instanceof File) {
            if (MAX_UPLOAD_FILE_SIZE && newFiles.size > MAX_UPLOAD_FILE_SIZE) {
                setChooseFilesError(
                    'File size must be less than ' + formatFileSize(MAX_UPLOAD_FILE_SIZE, fileMetadata)
                );
            } else {
                setChosenFiles([newFiles]);
            }
        } else if (newFiles instanceof Array) {
            if (MAX_UPLOAD_FILE_SIZE) {
                const numFilesAttempted = newFiles.length;
                newFiles = newFiles.filter((file) => file.size <= MAX_UPLOAD_FILE_SIZE);
                if (newFiles.length < numFilesAttempted) {
                    setChooseFilesError(
                        'File size must be less than ' + formatFileSize(MAX_UPLOAD_FILE_SIZE, fileMetadata)
                    );
                }
            }
            setChosenFiles(newFiles);
        }
    };

    const uploadFile = async (
        files: File[],
        index: number,
        postPolicyFields: { [key: string]: string },
        bucketUrl: string
    ) => {
        let postData = new FormData();
        for (const [key, entry] of Object.entries(postPolicyFields)) {
            postData.append(key, entry);
        }
        postData.append('file', files[index]);
        const data = {
            method: 'POST',
            body: postData
        };
        const response = await fetch(bucketUrl, data);
        if (200 <= response.status && response.status < 300) {
            setUploadProgress((uploadProgress) => uploadProgress + Math.ceil((1 / chosenFiles.length) * 100));
        } else {
            setFilesFailedToUpload((filesFailedToUpload) => [...filesFailedToUpload, files[index].name]);
        }
        if (index === chosenFiles.length - 1) {
            setAllFilesAttempted(true);
        }
    };

    const postResource = async (endpoint: string, params = {}) => {
        try {
            token = await generateToken();
            const response = await API.post(API_NAME, endpoint, {
                headers: {
                    Authorization: token
                },
                body: params
            });
            return response;
        } catch (err) {
            console.error(err);
        }
    };

    const handleUploadClick = async () => {
        setProgressStatus(Status.IN_PROGRESS);
        setFilesFailedToUpload((filesFailedToUpload) => []);
        setUploadResultMessage('');
        setUploadProgress(0);
        setShowProgressBar(true);
        setAllFilesAttempted(false);

        const userName = await getUsername();
        let promiseArray = [];
        for (let i = 0; i < chosenFiles.length; i++) {
            let apiRequest = postResource(`document`, {
                userId: userName,
                caseId: selectedCaseOption.value,
                caseName: selectedCaseOption.label,
                fileName: chosenFiles[i].name,
                fileExtension: '.' + chosenFiles[i].name.split('.').pop(),
                documentType: documentTypeList[i],
                tagging: `<Tagging><TagSet><Tag><Key>userId</Key><Value>${userName}</Value></Tag></TagSet></Tagging>`
            });
            promiseArray.push(apiRequest);
        }

        await Promise.allSettled(promiseArray).then(async (results) => {
            results.forEach((result, i) => {
                if (result.status === 'fulfilled') {
                    if (result.value === undefined) {
                        setFilesFailedToUpload((filesFailedToUpload) => [...filesFailedToUpload, chosenFiles[i].name]);
                        setUploadResultMessage((uploadResultMessage) => 'File upload failed');
                        setProgressStatus(Status.ERROR);
                    } else {
                        uploadFile(chosenFiles, i, result.value.fields, result.value.url);
                    }
                }
            });
        });
        setRefreshCaseData(true);
        setChosenFiles([]);
    };

    const allowChooseFiles = () => {
        if (selectedCaseOption.value !== undefined && caseMap[selectedCaseOption.value] !== undefined) {
            return (
                chosenFiles.length <
                props.workflowConfig.NumRequiredDocuments - caseMap[selectedCaseOption.value].caseDocuments.length
            );
        }
        return false;
    };

    const updateDocumentTypeList = (index: number, value: string | null) => {
        const updatedList = [...documentTypeList];
        if (index >= updatedList.length) {
            updatedList.push(value!);
        } else if (value === null) {
            updatedList.splice(index, 1);
        } else {
            updatedList[index] = value;
        }
        setDocumentTypeList(updatedList);
    };

    const allowUploadFiles = () => {
        if (selectedCaseOption.value !== undefined && caseMap[selectedCaseOption.value] !== undefined) {
            return (
                chosenFiles.length > 0 &&
                caseMap[selectedCaseOption.value].caseDocuments.length + chosenFiles.length <=
                    props.workflowConfig.NumRequiredDocuments
            );
        }
        return false;
    };

    React.useEffect(() => {
        if (uploadProgress < 100 && allFilesAttempted) {
            setProgressStatus(Status.ERROR);
            setUploadResultMessage((uploadResultMessage) => 'File upload failed');
        } else if (uploadProgress >= 100 && allFilesAttempted) {
            setProgressStatus(Status.SUCCESS);
            setUploadResultMessage((uploadResultMessage) => 'Files uploaded sucessfully');
        } else if (uploadProgress > 0 && uploadProgress < 100 && !allFilesAttempted) {
            setProgressStatus(Status.IN_PROGRESS);
        }
    }, [uploadProgress, progressStatus, allFilesAttempted]);

    return (
        <div>
            <AppLayout
                contentType="form"
                content={
                    <ContentLayout
                        header={
                            <SpaceBetween size="m">
                                <Header variant="h1">Upload Documents</Header>
                            </SpaceBetween>
                        }
                        data-testid="doc-upload-view-contentlayout"
                    >
                        <form onSubmit={(event) => event.preventDefault()}>
                            <Form
                                actions={
                                    <SpaceBetween direction="horizontal" size="xs">
                                        {showProgressBar && (
                                            <ProgressBar
                                                value={uploadProgress}
                                                additionalInfo={uploadResultMessage}
                                                variant="key-value"
                                                label="Document Upload Status"
                                                status={progressStatus}
                                                data-testid="upload-document-progress-bar"
                                            />
                                        )}
                                        <span>&nbsp;&nbsp;&nbsp;&nbsp;</span>
                                        <Button
                                            data-testid="upload-documents-button"
                                            variant="primary"
                                            onClick={() => handleUploadClick()}
                                            disabled={!allowUploadFiles()}
                                        >
                                            Upload Documents
                                        </Button>
                                        <Button variant="link" onClick={() => navigate(`/`)}>
                                            Done
                                        </Button>
                                    </SpaceBetween>
                                }
                                errorIconAriaLabel="Error"
                            >
                                <Container>
                                    <SpaceBetween direction="vertical" size="l">
                                        <FormField
                                            label="Case name"
                                            i18nStrings={{ errorIconAriaLabel: 'Error' }}
                                            data-testid="case-name-field"
                                        >
                                            <Select
                                                selectedOption={selectedCaseOption}
                                                onChange={({ detail }) => setSelectedCaseOption(detail.selectedOption)}
                                                options={casesOptions}
                                                filteringType="auto"
                                                selectedAriaLabel="Selected"
                                                data-testid="upload-doc-case-select"
                                            />
                                        </FormField>
                                        <FileUpload
                                            label="Files to upload"
                                            accept=".png,.pdf,.jpg,.jpeg"
                                            fileMetadata={fileMetadata}
                                            multiple={allowMultipleFiles}
                                            value={chosenFiles}
                                            description={
                                                'Please choose the files you wish to upload to the selected case'
                                            }
                                            disabled={!allowChooseFiles()}
                                            errorText={chooseFilesError}
                                            data-testid="file-upload-field"
                                            onChange={({ detail }) => handleFileSelection(detail.value)}
                                            workflowConfig={props.workflowConfig}
                                            documentTypeList={documentTypeList}
                                            setDocumentTypeList={updateDocumentTypeList}
                                            remainingAllowedDocs={selectedCaseRemainingAllowedDocs}
                                        ></FileUpload>
                                    </SpaceBetween>
                                </Container>
                            </Form>
                        </form>
                        {uploadResultMessage === 'File upload failed' && filesFailedToUpload.length > 0 && (
                            <Alert
                                statusIconAriaLabel="Error"
                                type="error"
                                header="Failed to upload the following files: "
                            >
                                {filesFailedToUpload.map((fileName, i) => (
                                    <li key={fileName + i}>{fileName}</li> // NOSONAR - fix to make displayed file names unique to be implemented in the future
                                ))}
                            </Alert>
                        )}
                    </ContentLayout>
                }
                headerSelector="#header"
                navigationHide
                toolsHide
                date-testid="doc-upload-view-applayout"
            />
        </div>
    );
}
