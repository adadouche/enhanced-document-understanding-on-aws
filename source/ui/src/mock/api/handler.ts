// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

import { rest } from 'msw';

export const MOCK_CONFIG = {
    ApiEndpoint: 'https://jsonplaceholder.typicode.com/',
    AWS_REGION: 'us-east-1',
    USER_POOL_ID: 'test_user_pool_id',
    USER_POOL_CLIENT_ID: 'testClientId',
    DATA_BUCKET_NAME: 'test-bucket'
};

export const getSignedUrlResponse = {
    downloadUrl: `${MOCK_CONFIG.ApiEndpoint}fake-url`
};

const getDocumentResponse = {
    key: `${MOCK_CONFIG.ApiEndpoint}/fake-key`
};

const getCasesResponse = {
    Count: 4,
    ScannedCount: 4,
    Items: [
        {
            CASE_ID: { S: 'fake-case-id-1' },
            CASE_NAME: { S: 'fake-case-name-1' },
            DOCUMENT_ID: { S: '0000' },
            STATUS: { S: 'success' },
            USER_ID: { S: 'fake-user-id' },
            DOC_COUNT: { N: '1' },
            CREATION_TIMESTAMP: { 'S': '2000-01-01T00:00:00.000Z' }
        },
        {
            CASE_ID: { S: 'fake-case-id-2' },
            CASE_NAME: { S: 'fake-case-name-2' },
            DOCUMENT_ID: { S: '0000' },
            STATUS: { S: 'success' },
            USER_ID: { S: 'fake-user-id' },
            DOC_COUNT: { N: '0' },
            CREATION_TIMESTAMP: { 'S': '2000-01-01T00:00:00.000Z' }
        },
        {
            CASE_ID: { S: 'fake-case-id-3' },
            CASE_NAME: { S: 'fake-case-name-3' },
            DOCUMENT_ID: { S: '0000' },
            STATUS: { S: 'success' },
            USER_ID: { S: 'fake-user-id' },
            DOC_COUNT: { N: '0' },
            CREATION_TIMESTAMP: { 'S': '2000-01-01T00:00:00.000Z' }
        },
        {
            CASE_ID: { S: 'fake-case-id-4' },
            CASE_NAME: { S: 'fake-case-name-4' },
            DOCUMENT_ID: { S: '0000' },
            STATUS: { S: 'success' },
            USER_ID: { S: 'fake-user-id' },
            DOC_COUNT: { N: '0' },
            CREATION_TIMESTAMP: { 'S': '2000-01-01T00:00:00.000Z' }
        }
    ]
};

const getDocumentByCaseIdResponse = {
    Count: 1,
    ScannedCount: 1,
    Items: [
        {
            BUCKET_NAME: { S: 'fake-bucket-name-1' },
            CASE_ID: { S: 'fake-case-id-1' },
            CASE_NAME: { S: 'fake-case-name-1' },
            DOCUMENT_ID: { S: 'fake-doc-id1' },
            DOCUMENT_TYPE: { S: 'passport' },
            S3_KEY: {
                S: 'fake-s3-key.jpg'
            },
            UPLOADED_FILE_EXTENSION: { S: 'jpg' },
            UPLOADED_FILE_NAME: { S: 'fake-file-name' },
            USER_ID: { S: 'fake-user-id' },
            CREATION_TIMESTAMP: { 'S': '2000-01-01T00:00:00.000Z' }
        }
    ]
};

export const handlers = [
    rest.get('/runtimeConfig.json', (_, res, ctx) => {
        return res(ctx.status(200), ctx.text(JSON.stringify(MOCK_CONFIG)));
    }),
    rest.get(`${MOCK_CONFIG.ApiEndpoint}cases`, (req, res, ctx) => {
        let response: any = getCasesResponse;
        if (req.params) {
            response = {
                ...getCasesResponse,
                LastEvaluatedKey: true
            };
        }
        return res(ctx.status(200), ctx.text(JSON.stringify(response)));
    }),
    rest.get(`${MOCK_CONFIG.ApiEndpoint}case/:caseId`, (_, res, ctx) => {
        return res(ctx.status(200), ctx.text(JSON.stringify(getDocumentByCaseIdResponse)));
    }),
    rest.get(`${MOCK_CONFIG.ApiEndpoint}document/:caseId/:documentId`, (req, res, ctx) => {
        return res(ctx.status(200), ctx.text(JSON.stringify(getDocumentResponse)));
    }),
    rest.get(`${MOCK_CONFIG.ApiEndpoint}document/download`, (_, res, ctx) => {
        return res(ctx.status(200), ctx.text(JSON.stringify(getSignedUrlResponse)));
    }),
    rest.get(`${MOCK_CONFIG.ApiEndpoint}inferences/:caseId/:documentId`, (_, res, ctx) => {
        return res(ctx.status(200), ctx.text(JSON.stringify(['textract-detectText', 'entity-medical-locations'])));
    }),
    rest.get(`${MOCK_CONFIG.ApiEndpoint}inferences/:caseId/:documentId/:inferenceType`, (req, res, ctx) => {
        return res(ctx.status(200), ctx.text(JSON.stringify({})));
    }),
    rest.post(`${MOCK_CONFIG.ApiEndpoint}document`, (_, res, ctx) => {
        return res(
            ctx.status(200),
            ctx.text(
                JSON.stringify({
                    fields: [
                        {
                            key: 'test-entry'
                        }
                    ],
                    url: 'signed-url'
                })
            )
        );
    }),
    rest.all('/signed-url', (req, res, ctx) => {
        return res(ctx.status(403));
    })
];
