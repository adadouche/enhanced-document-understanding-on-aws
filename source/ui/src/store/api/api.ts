/**********************************************************************************************************************
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.                                                *
 *                                                                                                                    *
 *  Licensed under the Apache License, Version 2.0 (the License). You may not use this file except in compliance      *
 *  with the License. A copy of the License is located at                                                             *
 *                                                                                                                    *
 *      http://www.apache.org/licenses/LICENSE-2.0                                                                    *
 *                                                                                                                    *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES *
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions    *
 *  and limitations under the License.                                                                                *
 *********************************************************************************************************************/

import { BaseQueryFn, createApi, FetchArgs, fetchBaseQuery, FetchBaseQueryError } from '@reduxjs/toolkit/query/react';
import { generateToken, getRuntimeConfig } from '../../utils/apiHelper';

export const dynamicBaseQuery: BaseQueryFn<string | FetchArgs, unknown, FetchBaseQueryError> = async (
    args: any,
    api: any,
    extraOptions: any
) => {
    const runtimeConfig = await getRuntimeConfig();
    const rawBaseQuery = fetchBaseQuery({
        baseUrl: runtimeConfig.ApiEndpoint,
        prepareHeaders: async (headers: any) => {
            if (process.env.NODE_ENV !== 'test') {
                const token = await generateToken();
                headers.set('Authorization', `${token}`);
            }
            return headers;
        }
    });
    return rawBaseQuery(args, api, extraOptions);
};
export const api = createApi({
    reducerPath: 'api',
    baseQuery: dynamicBaseQuery,
    tagTypes: ['Cases'],
    endpoints: () => ({
       
    }),
    refetchOnMountOrArgChange: true
});

