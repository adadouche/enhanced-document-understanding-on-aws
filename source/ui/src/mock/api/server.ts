// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

import { handlers } from "./handler";
import { setupServer } from "msw/node";
export const server = setupServer(...handlers);