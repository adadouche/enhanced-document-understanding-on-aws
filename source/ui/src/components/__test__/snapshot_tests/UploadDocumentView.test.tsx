/**********************************************************************************************************************
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.                                                *
 *                                                                                                                    *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance    *
 *  with the License. A copy of the License is located at                                                             *
 *                                                                                                                    *
 *      http://www.apache.org/licenses/LICENSE-2.0                                                                    *
 *                                                                                                                    *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES *
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions    *
 *  and limitations under the License.                                                                                *
 *********************************************************************************************************************/

import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';

import { MemoryRouter, Route, Routes } from 'react-router-dom';

import { API } from '@aws-amplify/api';
import UploadDocumentView from '../../UploadDocumentView';

const mockAPI = {
    get: jest.fn(),
    post: jest.fn()
};
jest.mock('@aws-amplify/api');
API.get = mockAPI.get;
API.post = mockAPI.post;

beforeEach(() => {
    mockAPI.post.mockReset();
});

jest.mock('@cloudscape-design/components', () => {
    const Components = jest.genMockFromModule('@cloudscape-design/components') as any;
    for (const componentName of Object.keys(Components)) {
        Components[componentName] = componentName;
    }
    return Components;
});

test('Snapshot test', async () => {
    const workflowConfig = {
        NumRequiredDocuments: 1,
        UniqueDocumentTypes: ['generic'],
        WorkflowConfigName: 'default'
    };

    const tree = renderer
        .create(
            <MemoryRouter initialEntries={['/uploadDocument']}>
                <Routes>
                    <Route
                        path="/uploadDocument"
                        element={
                            <UploadDocumentView
                                casesList={[{ name: 'example-case1' }]}
                                caseName="example-case1"
                                setCasesList={jest.fn()}
                                workflowConfig={workflowConfig}
                                selectedCaseRemainingRequiredDocs={{}}
                            />
                        }
                    />
                </Routes>
            </MemoryRouter>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
