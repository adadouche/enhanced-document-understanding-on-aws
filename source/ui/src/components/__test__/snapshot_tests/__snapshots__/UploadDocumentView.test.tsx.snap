// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test 1`] = `
<div>
  <AppLayout
    content={
      <ContentLayout
        data-testid="doc-upload-view-contentlayout"
        header={
          <SpaceBetween
            size="m"
          >
            <Header
              variant="h1"
            >
              Upload Documents
            </Header>
          </SpaceBetween>
        }
      >
        <form
          onSubmit={[Function]}
        >
          <Form
            actions={
              <SpaceBetween
                direction="horizontal"
                size="xs"
              >
                <span>
                      
                </span>
                <Button
                  data-testid="upload-documents-button"
                  disabled={true}
                  onClick={[Function]}
                  variant="primary"
                >
                  Upload Documents
                </Button>
                <Button
                  onClick={[Function]}
                  variant="link"
                >
                  Done
                </Button>
              </SpaceBetween>
            }
            errorIconAriaLabel="Error"
          >
            <Container>
              <SpaceBetween
                direction="vertical"
                size="l"
              >
                <FormField
                  data-testid="case-name-field"
                  i18nStrings={
                    Object {
                      "errorIconAriaLabel": "Error",
                    }
                  }
                  label="Case name"
                >
                  <Select
                    data-testid="upload-doc-case-select"
                    filteringType="auto"
                    onChange={[Function]}
                    options={
                      Array [
                        Object {
                          "description": "Case Id: undefined",
                          "disabled": true,
                          "label": "example-case1",
                          "value": "undefined",
                        },
                      ]
                    }
                    selectedAriaLabel="Selected"
                    selectedOption={
                      Object {
                        "label": "example-case1",
                        "value": undefined,
                      }
                    }
                  />
                </FormField>
                <ForwardRef(FileUploadComponent)
                  accept=".png,.pdf,.jpg,.jpeg"
                  data-testid="file-upload-field"
                  description="Please choose the files you wish to upload to the selected case"
                  disabled={true}
                  documentTypeList={Array []}
                  errorText=""
                  fileMetadata={
                    Object {
                      "lastModified": true,
                      "name": true,
                      "size": "MB",
                      "thumbnail": true,
                      "type": true,
                    }
                  }
                  label="Files to upload"
                  multiple={true}
                  onChange={[Function]}
                  setDocumentTypeList={[Function]}
                  value={Array []}
                  workflowConfig={
                    Object {
                      "NumRequiredDocuments": 1,
                      "UniqueDocumentTypes": Array [
                        "generic",
                      ],
                      "WorkflowConfigName": "default",
                    }
                  }
                />
              </SpaceBetween>
            </Container>
          </Form>
        </form>
      </ContentLayout>
    }
    contentType="form"
    date-testid="doc-upload-view-applayout"
    headerSelector="#header"
    navigationHide={true}
    toolsHide={true}
  />
</div>
`;
